File: code_snippets/go/develop/servinglambda.go
package develop

import (
	"log"

	"github.com/aws/aws-lambda-go/lambda"
	restate "github.com/restatedev/sdk-go"
	server "github.com/restatedev/sdk-go/server"
)

type MyService struct{}
type MyVirtualObject struct{}
type MyWorkflow struct{}

func serveLambda() {
	// <start_lambda>
	handler, err := server.NewRestate().
		Bind(restate.Reflect(MyService{})).
		Bind(restate.Reflect(MyVirtualObject{})).
		Bind(restate.Reflect(MyWorkflow{})).
		Bidirectional(false).
		LambdaHandler()
	if err != nil {
		log.Fatal(err)
	}
	lambda.Start(handler)
	// <end_lambda>

}

File: code_snippets/ts/src/develop/serving_lambda.ts
const myService = restate.service({
  name: "MyService",
  handlers: {},
});

const myVirtualObject = restate.object({
  name: "MyVirtualObject",
  handlers: {},
});

const myWorkflow = restate.workflow({
  name: "MyWorkflow",
  handlers: { run: async () => {} },
});

// <start_lambda>
import * as restate from "@restatedev/restate-sdk/lambda";
export const handler = restate
  .endpoint()
  .bind(myService)
  .bind(myVirtualObject)
  .bind(myWorkflow)
  // !mark
  .handler();
// <end_lambda>
